### PR TITLE
feat: add metric counting using OTL

### PR DESCRIPTION
**Pull request by stephhuynh18**

### CODE REVIEW

File: Dockerfile.launchContract [hunk 1]
```diff
@@ -3,6 +3,6 @@ FROM ghcr.io/foundry-rs/foundry as contractLauncher
 # Build and test the source code
 COPY ./contracts .
 RUN forge install foundry-rs/forge-std --no-git
-RUN forge install openzeppelin/openzeppelin-contracts --no-git
-RUN forge build --use 0.8.17
+RUN forge install openzeppelin/openzeppelin-contracts@v4.8.0 --no-git
+RUN forge build --use 0.8.13
 ENTRYPOINT forge create --contracts ./src/CeramicAnchorServiceV2.sol --private-key $ETH_WALLET_PK --rpc-url $ETH_RPC_URL CeramicAnchorServiceV2
```
Comment (smrz2001) [inline Dockerfile.launchContract]: Mind adding a newline at the end of this file?

File: models/constants.go [hunk 2]
```diff
@@ -20,3 +20,8 @@ const ServiceName = "cas-scheduler"
 const (
 	Env_EnvTag = "ENV_TAG"
 )
+
+// Metrics
+// Counts
+const CreatedBatchMetricName = "created_batch"
```
Comment (smrz2001) [inline models/constants.go]: Should we make these into an enum?
```
type MetricName string

const (
	MetricName_CreatedBatch    MetricName = "created_batch"
	MetricName_ReplacedRequest MetricName = "replaced_request"
)

```

Comment (smrz2001) [reply models/constants.go]: Do we need additional metrics?
- Ingress requests for the Validation service so we can see how much load we're receiving
- Ingress requests to the Batching service so we can see how many requests got past deduplication
- Avg. number of requests in a Batch so we can see the pattern of batch sizes over time under different loads
- Avg. batch linger (need to check if this is possible to measure) so we can see the time it's taking for batches to be issued.
- Number of anchor jobs created by the Worker service
- Number of failures received by the Failure Handler
- Others?

Don't have to implement all these in this PR though we should track this list somewhere and add them in subsequent PRs.

File: common/metrics/otl.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	"github.com/ceramicnetwork/go-cas/models"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+	"go.opentelemetry.io/otel/exporters/stdout/stdoutmetric"
+	"go.opentelemetry.io/otel/metric"
+	sdk "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+)
+
+type OTLMetricService struct {
```
Comment (smrz2001) [inline common/metrics/otl.go]: minor: Should this be named `OtlMetricService`?

Comment (gvelez17) [reply common/metrics/otl.go]: i think not in case we change the internals later

File: common/metrics/otl.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	resources, err := resource.New(ctx,
+		resource.WithFromEnv(), // pull attributes from OTEL_RESOURCE_ATTRIBUTES and OTEL_SERVICE_NAME environment variables
+		resource.WithAttributes(attribute.String("environment", os.Getenv("ENV"))),
+	)
+	if err != nil {
+		return &OTLMetricService{}, err
+	}
+
+	var exporter sdk.Exporter
+	collectorUrl := os.Getenv("COLLECTOR_URL")
```
Comment (gvelez17) [inline common/metrics/otl.go]: the actual env variable is COLLECTOR_HOST

Comment (gvelez17) [reply common/metrics/otl.go]: we normally pass this to the observability package from the caller

File: common/metrics/otl.go [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+			return &OTLMetricService{}, err
+		}
+	}
+
+	meterProvider := sdk.NewMeterProvider(
+		sdk.WithResource(resources),
+		sdk.WithReader(sdk.NewPeriodicReader(exporter)),
+	)
+
+	meter := meterProvider.Meter("github.com/ceramicnetwork/go-cas/common/metrics")
```
Comment (gvelez17) [inline common/metrics/otl.go]: maybe just 'go-cas'

later we can move this to a go observability repo and use caller again

File: common/metrics/otl.go (hunk 3 above)
Comment (smrz2001) [reply common/metrics/otl.go]: I'm not sure I understand. Was just recommending changing `OTL` -> `Otl` to preserve camel casing. Should we not?

File: common/metrics/otl.go [hunk 6]
```diff
@@ -0,0 +1,77 @@
+package metrics
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+
+	"github.com/ceramicnetwork/go-cas/models"
```
Comment (smrz2001) [inline common/metrics/otl.go]: minor: Move this line to a new section after the imports below separated by a newline

File: models/constants.go (hunk 2 above)
Comment (smrz2001) [reply models/constants.go]: Also question for @gvelez17 here - when we replace the CASv2 Scheduler with the CASv5 Scheduler, will we lose any metrics you might have added to the old Scheduler? If so, then we probably need to make sure that at least those metrics, if still useful in the new design, are retained in the new code.

Comment (stephhuynh18) [reply models/constants.go]: I'll make a ticket

**Review by smrz2001 — APPROVED** [review]
🚀

**Pull request merged**